### DISCUSSION TITLE
How to use existing PDO instance in doctrine/dbal 3.3.*

### DISCUSSION THREAD

**Discussion by jakublabno** [doctrine/doctrine2#9464] [Support Questions] [answered]
`BC BREAK User-provided PDO instance is no longer supported
In order to share the same PDO instances between DBAL and other components, initialize the connection in DBAL and access it using`

How to use existing instance without creating new connection to database?

Before I used `DriverManager::getConnection(['pdo' => PDO_INSTANCE]);` but it does not work anymore.

Using `'pdo'` option results in mysql error `No such file or directory`

**Comment by jakublabno** [comment]
For now I've used own driver, but it should not be a solution…I think

**Comment by derrabus** [comment, marked as answer]
You cannot. In order to share the same `PDO` instance between DBAL and other components, initialize the connection in DBAL and access it using `Connection::getNativeConnection()`.

**Reply by jakublabno** [reply]
> You cannot. In order to share the same `PDO` instance between DBAL and other components, initialize the connection in DBAL and access it using `Connection::getNativeConnection()`.

Maybe there should be an exception thrown when pdo parameter is passed, this exception strict from database server is quite confusing...

**Reply by derrabus** [reply]
Sure. If you have an idea how to improve the developer experience in your case, please submit a PR to the DBAL repository and we consider it. 🙂

**Reply by ramsey** [reply]
We created a library that relies on the ability to pass an existing PDO connection into it, and internally, the library uses this PDO connection with Doctrine in the same way @jakublabno describes. When trying to upgrade to Doctrine 3.x, I ran into this issue, which led me here.

To be clear, we don't want to get the native connection from Doctrine. We want to give Doctrine an existing native connection that it can use with the ORM.

From your answer, it sounds like this is no longer possible?

**Reply by beberlei** [reply]
You can build a driver yourself that does this. Afaik Laravel does that so the code is out there somewhere

**Reply by jakublabno** [reply]
Yes, I've built my own driver, it's very simple to achieve that 🙂

**Reply by ramsey** [reply]
Okay. That's what I've done, then.
